keg_elements/db/utils.py

```diff
@@ -178,18 +178,28 @@ def find_child(self, data):
     @no_autoflush
     def update(self):
         """Update the objects associated with the entity"""
+        to_add = []
+        to_edit = []
         for record in self.data:
             child = self.find_child(record)
             state = sa.inspect(child) if child else None
 
             if state and state.persistent:
-                child.edit(_commit=False, **record)
+                to_edit.append((child, record))
             else:
-                child = self.child_cls.add(_commit=False, **record)
-                self.collection.append(child)
+                to_add.append(record)
 
             self.keep_children.add(child)
+
         self.remove_unmodified()
+        session_flush()
```

Comment:
Might be good to wrap this entire function in [`might_commit`](https://github.com/level12/keg-elements/blob/master/keg_elements/db/mixins.py#L20-L21) / `might_flush` so the dev can control it.

Reply:
If only works if we flush at this particular moment so `might_flush` won't do it.